libtenzir/builtins/operators/to_hive.cpp

```diff
@@ -0,0 +1,428 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "tenzir/detail/zip_iterator.hpp"
+#include "tenzir/si_literals.hpp"
+#include "tenzir/tql2/eval.hpp"
+#include "tenzir/tql2/plugin.hpp"
+
+#include <ranges>
+
+// TODO: This implementation is a rough sketch and needs some cleanup eventually.
+
+namespace tenzir::plugins::to_hive {
+
+namespace {
+
+struct operator_args {
+  friend auto inspect(auto& f, operator_args& x) -> bool {
+    return f.object(x).fields(f.field("uri", x.uri), f.field("by", x.by),
+                              f.field("extension", x.extension),
+                              f.field("writer", x.writer),
+                              f.field("timeout", x.timeout),
+                              f.field("max_size", x.max_size));
+  }
+
+  std::string uri;
+  std::vector<ast::simple_selector> by;
+  std::string extension;
+  pipeline writer;
+  duration timeout{};
+  uint64_t max_size{};
+};
+
+// TODO: Don't we have this already?
+auto is_empty(const table_slice& x) -> bool {
+  return x.rows() == 0;
+}
+
+auto is_empty(const chunk_ptr& x) -> bool {
+  return not x || x->size() == 0;
+}
+
+auto is_empty(std::monostate) -> bool {
+  return true;
+}
```

Comment:
We have a `size(…)` free function, but that's not implemented for `std::monostate`.